Don't install dependency links at install time for the binfmt services, use
[Install] blocks so that they get created when the service is enabled like a
traditional service.

The [Install] blocks were rejected upstream as they don't have a way to "enable"
it on install without static symlinks which can't be disabled, only masked.  We
however can do that in a postinst.

Upstream-Status: Denied
Signed-off-by: Ross Burton <ross.burton@intel.com>

Index: git/Makefile.am
===================================================================
--- git.orig/Makefile.am
+++ git/Makefile.am
@@ -4495,10 +4495,6 @@
 	$(prefix)/lib/binfmt.d \
 	$(sysconfdir)/binfmt.d
 
-SYSINIT_TARGET_WANTS += \
-	systemd-binfmt.service \
-	proc-sys-fs-binfmt_misc.automount
-
 endif
 
 EXTRA_DIST += \
Index: git/units/proc-sys-fs-binfmt_misc.automount
===================================================================
--- git.orig/units/proc-sys-fs-binfmt_misc.automount
+++ git/units/proc-sys-fs-binfmt_misc.automount
@@ -16,3 +16,6 @@
 
 [Automount]
 Where=/proc/sys/fs/binfmt_misc
+
+[Install]
+WantedBy=sysinit.target
Index: git/units/systemd-binfmt.service.in
===================================================================
--- git.orig/units/systemd-binfmt.service.in
+++ git/units/systemd-binfmt.service.in
@@ -11,6 +11,8 @@
 Documentation=https://www.kernel.org/doc/Documentation/binfmt_misc.txt
 DefaultDependencies=no
 Conflicts=shutdown.target
+Wants=proc-sys-fs-binfmt_misc.automount
+
 After=proc-sys-fs-binfmt_misc.automount
 Before=sysinit.target shutdown.target
 ConditionPathIsReadWrite=/proc/sys/
@@ -24,3 +26,6 @@
 Type=oneshot
 RemainAfterExit=yes
 ExecStart=@rootlibexecdir@/systemd-binfmt
+
+[Install]
+WantedBy=sysinit.target
